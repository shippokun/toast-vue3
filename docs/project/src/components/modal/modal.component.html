<!DOCTYPE html><html lang="en"><head><title>@code-dependency/view</title><meta charSet="UTF-8"/><meta name="description" content="visualize code dependency with graphviz."/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><meta http-equiv="X-UA-Compatible" content="ie=edge"/><script src="https://shippokun.github.io/toast-vue3/assets/scripts/react.production.min.js"></script><script src="https://shippokun.github.io/toast-vue3/assets/scripts/react-dom.production.min.js"></script><link href="https://shippokun.github.io/toast-vue3/assets/stylesheets/styles.0e433876.chunk.css" rel="stylesheet"/><script>window.__INITIAL_PROPS__ = {"isServer":true,"isStatic":true,"publicPath":"https://shippokun.github.io/toast-vue3/","selectedPathname":"src/components/modal/modal.component.tsx","publicPathname":"/project","pagePathname":"/toast-vue3/","workerUrl":"https://shippokun.github.io/toast-vue3/assets/scripts/full.render.js","assetBaseUrl":"https://shippokun.github.io/toast-vue3/assets","sourceType":"svg","svgElement":"<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n<!-- Generated by graphviz version 2.40.1 (20161225.0304)\n -->\n<!-- Title: dependency&#45;cruiser output Pages: 1 -->\n<svg width=\"255pt\" height=\"147pt\"\n viewBox=\"0.00 0.00 254.53 147.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n<g id=\"graph0\" class=\"graph\" transform=\"scale(1 1) rotate(0) translate(4 143)\">\n<title>dependency&#45;cruiser output</title>\n<polygon fill=\"#ffffff\" stroke=\"transparent\" points=\"-4,4 -4,-143 250.534,-143 250.534,4 -4,4\"/>\n<g id=\"clust1\" class=\"cluster\">\n<title>cluster_src</title>\n<path fill=\"#ffffff\" stroke=\"#000000\" stroke-width=\"2\" d=\"M20,-8C20,-8 145.5216,-8 145.5216,-8 151.5216,-8 157.5216,-14 157.5216,-20 157.5216,-20 157.5216,-119 157.5216,-119 157.5216,-125 151.5216,-131 145.5216,-131 145.5216,-131 20,-131 20,-131 14,-131 8,-125 8,-119 8,-119 8,-20 8,-20 8,-14 14,-8 20,-8\"/>\n<text text-anchor=\"middle\" x=\"82.7608\" y=\"-118.9\" font-family=\"Helvetica,sans-Serif\" font-weight=\"bold\" font-size=\"9.00\" fill=\"#000000\">src</text>\n</g>\n<g id=\"clust2\" class=\"cluster\">\n<title>cluster_src/components</title>\n<path fill=\"#ffffff\" stroke=\"#000000\" stroke-width=\"2\" d=\"M28,-16C28,-16 137.5216,-16 137.5216,-16 143.5216,-16 149.5216,-22 149.5216,-28 149.5216,-28 149.5216,-92 149.5216,-92 149.5216,-98 143.5216,-104 137.5216,-104 137.5216,-104 28,-104 28,-104 22,-104 16,-98 16,-92 16,-92 16,-28 16,-28 16,-22 22,-16 28,-16\"/>\n<text text-anchor=\"middle\" x=\"82.7608\" y=\"-91.9\" font-family=\"Helvetica,sans-Serif\" font-weight=\"bold\" font-size=\"9.00\" fill=\"#000000\">components</text>\n</g>\n<g id=\"clust3\" class=\"cluster\">\n<title>cluster_src/components/modal</title>\n<path fill=\"#ffffff\" stroke=\"#000000\" stroke-width=\"2\" d=\"M36,-24C36,-24 129.5216,-24 129.5216,-24 135.5216,-24 141.5216,-30 141.5216,-36 141.5216,-36 141.5216,-65 141.5216,-65 141.5216,-71 135.5216,-77 129.5216,-77 129.5216,-77 36,-77 36,-77 30,-77 24,-71 24,-65 24,-65 24,-36 24,-36 24,-30 30,-24 36,-24\"/>\n<text text-anchor=\"middle\" x=\"82.7608\" y=\"-64.9\" font-family=\"Helvetica,sans-Serif\" font-weight=\"bold\" font-size=\"9.00\" fill=\"#000000\">modal</text>\n</g>\n<!-- ./modal.presenter -->\n<g id=\"node1\" class=\"node\">\n<title>./modal.presenter</title>\n<path fill=\"#ffffcc\" stroke=\"#ff0000\" d=\"M240.3395,-50.3011C240.3395,-50.3011 171.7161,-50.3011 171.7161,-50.3011 168.6157,-50.3011 165.5154,-47.2007 165.5154,-44.1004 165.5154,-44.1004 165.5154,-37.8996 165.5154,-37.8996 165.5154,-34.7993 168.6157,-31.6989 171.7161,-31.6989 171.7161,-31.6989 240.3395,-31.6989 240.3395,-31.6989 243.4399,-31.6989 246.5402,-34.7993 246.5402,-37.8996 246.5402,-37.8996 246.5402,-44.1004 246.5402,-44.1004 246.5402,-47.2007 243.4399,-50.3011 240.3395,-50.3011\"/>\n<text text-anchor=\"middle\" x=\"206.0278\" y=\"-38.3\" font-family=\"Helvetica,sans-Serif\" font-size=\"9.00\" fill=\"#ff0000\">modal.presenter</text>\n</g>\n<!-- src/components/modal/modal.component.tsx -->\n<g id=\"node2\" class=\"node\">\n<title>src/components/modal/modal.component.tsx</title>\n<g id=\"a_node2\"><a xlink:href=\"src/components/modal/modal.component.tsx\" xlink:title=\"modal.component.tsx\">\n<path fill=\"#bbfeff\" stroke=\"#000000\" d=\"M127.0828,-50.3011C127.0828,-50.3011 38.4388,-50.3011 38.4388,-50.3011 35.3384,-50.3011 32.2381,-47.2007 32.2381,-44.1004 32.2381,-44.1004 32.2381,-37.8996 32.2381,-37.8996 32.2381,-34.7993 35.3384,-31.6989 38.4388,-31.6989 38.4388,-31.6989 127.0828,-31.6989 127.0828,-31.6989 130.1832,-31.6989 133.2835,-34.7993 133.2835,-37.8996 133.2835,-37.8996 133.2835,-44.1004 133.2835,-44.1004 133.2835,-47.2007 130.1832,-50.3011 127.0828,-50.3011\"/>\n<text text-anchor=\"middle\" x=\"82.7608\" y=\"-38.3\" font-family=\"Helvetica,sans-Serif\" font-size=\"9.00\" fill=\"#000000\">modal.component.tsx</text>\n</a>\n</g>\n</g>\n<!-- src/components/modal/modal.component.tsx&#45;&gt;./modal.presenter -->\n<g id=\"edge1\" class=\"edge\">\n<title>src/components/modal/modal.component.tsx&#45;&gt;./modal.presenter</title>\n<path fill=\"none\" stroke=\"#000000\" stroke-width=\"2\" stroke-opacity=\"0.200000\" d=\"M133.6099,-41C142.0655,-41 150.8007,-41 159.1518,-41\"/>\n<polygon fill=\"#000000\" fill-opacity=\"0.200000\" stroke=\"#000000\" stroke-width=\"2\" stroke-opacity=\"0.200000\" points=\"159.4522,-43.1001 165.4521,-41 159.4521,-38.9001 159.4522,-43.1001\"/>\n</g>\n</g>\n</svg>\n","rendererType":"server","filePathList":[{"source":"src/App.tsx"},{"source":"src/main.ts"},{"source":"src/shims-vue.d.ts"},{"source":"src/router.ts"},{"source":"src/libs/index.ts"},{"source":"src/libs/date/date.ts"},{"source":"src/libs/date/index.ts"},{"source":"src/services/index.ts"},{"source":"src/services/todo/index.ts"},{"source":"src/services/todo/todo.service.ts"},{"source":"src/layouts/todo/index.ts"},{"source":"src/layouts/todo/todo.layout.spec.tsx"},{"source":"src/layouts/todo/todo.layout.tsx"},{"source":"src/components/index.ts"},{"source":"src/components/vscode/vscode.component.tsx"},{"source":"src/components/vscode/index.ts"},{"source":"src/components/modal/index.ts"},{"source":"src/components/modal/modal.component.spec.tsx"},{"source":"src/components/modal/modal.presenter.ts"},{"source":"src/components/modal/modal.component.tsx"},{"source":"src/models/index.ts"},{"source":"src/models/testing/index.ts"},{"source":"src/models/todo/index.ts"},{"source":"src/models/todo/todo.model.ts"},{"source":"src/models/todo/testing/todo.mock.ts"},{"source":"src/models/todo/testing/index.ts"},{"source":"src/store/index.ts"},{"source":"src/store/store.ts"},{"source":"src/store/todo/index.ts"},{"source":"src/store/todo/actions/index.ts"},{"source":"src/store/todo/actions/todo.action.ts"},{"source":"src/store/todo/getters/index.ts"},{"source":"src/store/todo/getters/todo.getter.ts"},{"source":"src/store/todo/muitations/todo.mutation.ts"},{"source":"src/store/todo/muitations/index.ts"},{"source":"src/store/todo/facades/index.ts"},{"source":"src/store/todo/facades/todo.facade.ts"},{"source":"src/store/todo/states/index.ts"},{"source":"src/store/todo/states/todo.state.ts"},{"source":"src/pages/todo/todo-create/index.ts"},{"source":"src/pages/todo/todo-create/todo-create.page.tsx"},{"source":"src/pages/todo/todo-create/containers/index.ts"},{"source":"src/pages/todo/todo-create/containers/todo-create/todo-create.container.tsx"},{"source":"src/pages/todo/todo-create/containers/todo-create/index.ts"},{"source":"src/pages/todo/todo-create/components/index.ts"},{"source":"src/pages/todo/todo-create/components/todo-create/todo-create.component.tsx"},{"source":"src/pages/todo/todo-create/components/todo-create/todo-create.presenter.ts"},{"source":"src/pages/todo/todo-create/components/todo-create/index.ts"},{"source":"src/pages/todo/todo-edit/index.ts"},{"source":"src/pages/todo/todo-edit/todo-edit.page.tsx"},{"source":"src/pages/todo/todo-edit/components/index.ts"},{"source":"src/pages/todo/todo-edit/components/todo-edit/index.ts"},{"source":"src/pages/todo/todo-edit/components/todo-edit/todo-edit.presenter.ts"},{"source":"src/pages/todo/todo-edit/components/todo-edit/todo-edit.component.tsx"},{"source":"src/pages/todo/todo-edit/containers/index.ts"},{"source":"src/pages/todo/todo-edit/containers/todo-edit/index.ts"},{"source":"src/pages/todo/todo-edit/containers/todo-edit/todo-edit.container.tsx"},{"source":"src/pages/todo/todo-detail/index.ts"},{"source":"src/pages/todo/todo-detail/todo-detail.page.tsx"},{"source":"src/pages/todo/todo-detail/components/index.ts"},{"source":"src/pages/todo/todo-detail/components/todo-detail/index.ts"},{"source":"src/pages/todo/todo-detail/components/todo-detail/todo-detail.component.tsx"},{"source":"src/pages/todo/todo-detail/containers/index.ts"},{"source":"src/pages/todo/todo-detail/containers/todo-detail/index.ts"},{"source":"src/pages/todo/todo-detail/containers/todo-detail/todo-detail.container.tsx"},{"source":"src/pages/todo/todo-list/index.ts"},{"source":"src/pages/todo/todo-list/todo-list.page.tsx"},{"source":"src/pages/todo/todo-list/containers/index.ts"},{"source":"src/pages/todo/todo-list/containers/todo-list/index.ts"},{"source":"src/pages/todo/todo-list/containers/todo-list/todo-list.container.tsx"},{"source":"src/pages/todo/todo-list/components/index.ts"},{"source":"src/pages/todo/todo-list/components/todo-list/todo-list.component.spec.tsx"},{"source":"src/pages/todo/todo-list/components/todo-list/index.ts"},{"source":"src/pages/todo/todo-list/components/todo-list/todo-list.component.tsx"}]}</script></head><body><div id="root"><div style="background-color:#fafbfd;display:flex"><div style="margin-right:24px;min-width:192px;flex:0 1 0%"><div style="position:absolute;width:192px;bottom:0;top:8px;overflow-y:scroll"><nav><div id="." style="padding-left:0"><span style="padding-left:8px;padding-right:8px;line-height:1.5;font-size:14px;cursor:pointer;width:100%">@code-dependency</span></div></nav></div></div><main style="display:flex;flex:1 1 100%;flex-direction:column"><div style="display:flex;flex-direction:row;align-items:center;margin-bottom:16px"><h1 style="color:#343741;font-size:16px">src/components/modal/modal.component.tsx</h1></div><div style="padding:24px;box-shadow:0 2px 2px -1px rgba(152,162,179,.3), 0 1px 5px -2px rgba(152,162,179,.3);background-color:#FFF;border-radius:4px;border:1px solid #d3dae6;flex-grow:1"><div><div><?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.40.1 (20161225.0304)
 -->
<!-- Title: dependency&#45;cruiser output Pages: 1 -->
<svg width="255pt" height="147pt"
 viewBox="0.00 0.00 254.53 147.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 143)">
<title>dependency&#45;cruiser output</title>
<polygon fill="#ffffff" stroke="transparent" points="-4,4 -4,-143 250.534,-143 250.534,4 -4,4"/>
<g id="clust1" class="cluster">
<title>cluster_src</title>
<path fill="#ffffff" stroke="#000000" stroke-width="2" d="M20,-8C20,-8 145.5216,-8 145.5216,-8 151.5216,-8 157.5216,-14 157.5216,-20 157.5216,-20 157.5216,-119 157.5216,-119 157.5216,-125 151.5216,-131 145.5216,-131 145.5216,-131 20,-131 20,-131 14,-131 8,-125 8,-119 8,-119 8,-20 8,-20 8,-14 14,-8 20,-8"/>
<text text-anchor="middle" x="82.7608" y="-118.9" font-family="Helvetica,sans-Serif" font-weight="bold" font-size="9.00" fill="#000000">src</text>
</g>
<g id="clust2" class="cluster">
<title>cluster_src/components</title>
<path fill="#ffffff" stroke="#000000" stroke-width="2" d="M28,-16C28,-16 137.5216,-16 137.5216,-16 143.5216,-16 149.5216,-22 149.5216,-28 149.5216,-28 149.5216,-92 149.5216,-92 149.5216,-98 143.5216,-104 137.5216,-104 137.5216,-104 28,-104 28,-104 22,-104 16,-98 16,-92 16,-92 16,-28 16,-28 16,-22 22,-16 28,-16"/>
<text text-anchor="middle" x="82.7608" y="-91.9" font-family="Helvetica,sans-Serif" font-weight="bold" font-size="9.00" fill="#000000">components</text>
</g>
<g id="clust3" class="cluster">
<title>cluster_src/components/modal</title>
<path fill="#ffffff" stroke="#000000" stroke-width="2" d="M36,-24C36,-24 129.5216,-24 129.5216,-24 135.5216,-24 141.5216,-30 141.5216,-36 141.5216,-36 141.5216,-65 141.5216,-65 141.5216,-71 135.5216,-77 129.5216,-77 129.5216,-77 36,-77 36,-77 30,-77 24,-71 24,-65 24,-65 24,-36 24,-36 24,-30 30,-24 36,-24"/>
<text text-anchor="middle" x="82.7608" y="-64.9" font-family="Helvetica,sans-Serif" font-weight="bold" font-size="9.00" fill="#000000">modal</text>
</g>
<!-- ./modal.presenter -->
<g id="node1" class="node">
<title>./modal.presenter</title>
<path fill="#ffffcc" stroke="#ff0000" d="M240.3395,-50.3011C240.3395,-50.3011 171.7161,-50.3011 171.7161,-50.3011 168.6157,-50.3011 165.5154,-47.2007 165.5154,-44.1004 165.5154,-44.1004 165.5154,-37.8996 165.5154,-37.8996 165.5154,-34.7993 168.6157,-31.6989 171.7161,-31.6989 171.7161,-31.6989 240.3395,-31.6989 240.3395,-31.6989 243.4399,-31.6989 246.5402,-34.7993 246.5402,-37.8996 246.5402,-37.8996 246.5402,-44.1004 246.5402,-44.1004 246.5402,-47.2007 243.4399,-50.3011 240.3395,-50.3011"/>
<text text-anchor="middle" x="206.0278" y="-38.3" font-family="Helvetica,sans-Serif" font-size="9.00" fill="#ff0000">modal.presenter</text>
</g>
<!-- src/components/modal/modal.component.tsx -->
<g id="node2" class="node">
<title>src/components/modal/modal.component.tsx</title>
<g id="a_node2"><a xlink:href="src/components/modal/modal.component.tsx" xlink:title="modal.component.tsx">
<path fill="#bbfeff" stroke="#000000" d="M127.0828,-50.3011C127.0828,-50.3011 38.4388,-50.3011 38.4388,-50.3011 35.3384,-50.3011 32.2381,-47.2007 32.2381,-44.1004 32.2381,-44.1004 32.2381,-37.8996 32.2381,-37.8996 32.2381,-34.7993 35.3384,-31.6989 38.4388,-31.6989 38.4388,-31.6989 127.0828,-31.6989 127.0828,-31.6989 130.1832,-31.6989 133.2835,-34.7993 133.2835,-37.8996 133.2835,-37.8996 133.2835,-44.1004 133.2835,-44.1004 133.2835,-47.2007 130.1832,-50.3011 127.0828,-50.3011"/>
<text text-anchor="middle" x="82.7608" y="-38.3" font-family="Helvetica,sans-Serif" font-size="9.00" fill="#000000">modal.component.tsx</text>
</a>
</g>
</g>
<!-- src/components/modal/modal.component.tsx&#45;&gt;./modal.presenter -->
<g id="edge1" class="edge">
<title>src/components/modal/modal.component.tsx&#45;&gt;./modal.presenter</title>
<path fill="none" stroke="#000000" stroke-width="2" stroke-opacity="0.200000" d="M133.6099,-41C142.0655,-41 150.8007,-41 159.1518,-41"/>
<polygon fill="#000000" fill-opacity="0.200000" stroke="#000000" stroke-width="2" stroke-opacity="0.200000" points="159.4522,-43.1001 165.4521,-41 159.4521,-38.9001 159.4522,-43.1001"/>
</g>
</g>
</svg>
</div></div></div></main></div></div><script type="text/javascript" src="https://shippokun.github.io/toast-vue3/assets/scripts/vendor.77e3c27a6a.js"></script><script type="text/javascript" src="https://shippokun.github.io/toast-vue3/assets/scripts/styles.e302b899f5.js"></script><script type="text/javascript" src="https://shippokun.github.io/toast-vue3/assets/scripts/application.7258f739f33c3c17ad1f.js"></script></body></html>