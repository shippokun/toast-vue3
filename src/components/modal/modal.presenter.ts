import { Ref, ref } from "vue";

export const useModal = (isShow: Ref<boolean> | undefined) => {
  const show = isShow || ref(false);
  const toggle = () => {
    show.value = !show.value;
  };

  return {
    show,
    toggle,
  } as const;
};
