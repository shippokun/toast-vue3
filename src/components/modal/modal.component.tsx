import { defineComponent, Teleport, PropType, renderSlot } from "vue";

import { useModal } from "./modal.presenter";

type Props = {
  isShow?: boolean;
};

export const ModalComponent = defineComponent({
  name: "ModalComponent",
  props: {
    isShow: {
      type: Boolean as PropType<Props["isShow"]>,
      default: false,
    },
  },
  setup(props: Props, { slots }) {
    const { show, toggle } = useModal(props.isShow);

    const slot = renderSlot(slots, "default");

    return () => (
      <Teleport to="body">
        {show.value && (
          <div class="modal is-active">
            <div class="modal-background"></div>
            <div class="modal-content">
              <div class="box">{slot}</div>
            </div>
            <button
              class="modal-close is-large"
              aria-label="close"
              id="closeButton"
              onClick={() => toggle()}
            ></button>
          </div>
        )}
      </Teleport>
    );
  },
});
