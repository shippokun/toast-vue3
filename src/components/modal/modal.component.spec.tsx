import { mount } from "@vue/test-utils";
import { ref } from "vue";
import { ModalComponent } from "./modal.component";

const createContainer = (tag = "div") => {
  const container = document.createElement(tag);
  document.body.appendChild(container);
  return container;
};

import { useModal } from "./modal.presenter";
jest.mock("./modal.presenter");

describe("modal", () => {
  const initBody = document.body.innerHTML;
  const toggleMock = jest.fn();
  beforeEach(() => {
    document.body.innerHTML = initBody;
  });
  it("isShowがfalseのとき、なにもrenderしないこと", () => {
    (useModal as jest.Mock).mockImplementation(() => {
      return {
        toggle: toggleMock,
        show: ref(false),
      };
    });
    mount(ModalComponent, {
      attachTo: createContainer(),
    });

    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it("isShowがtrueのとき、renderすること", () => {
    (useModal as jest.Mock).mockImplementation(() => {
      return {
        toggle: toggleMock,
        show: ref(true),
      };
    });

    const mockSlot = "test";
    mount(ModalComponent, {
      props: {
        isShow: ref(true),
      },
      attachTo: createContainer(),
      slots: {
        default: mockSlot,
      },
    });

    expect(document.body.innerHTML).toMatchSnapshot();
    expect(
      document.body.getElementsByClassName("modal is-active")
    ).toBeDefined();
    expect(toggleMock).not.toHaveBeenCalled();

    (document.getElementById("closeButton") as HTMLButtonElement).click();
    expect(toggleMock).toHaveBeenCalled();

    expect(document.body.getElementsByClassName("box")[0].innerHTML).toBe(
      mockSlot
    );
  });
});
