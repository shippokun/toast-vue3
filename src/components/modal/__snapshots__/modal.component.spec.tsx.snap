// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal isShowがfalseのとき、なにもrenderしないこと 1`] = `
<div>
  <div data-v-app="">
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>
<!---->
`;

exports[`modal isShowがtrueのとき、renderすること 1`] = `
<div>
  <div data-v-app="">
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>
<div class="modal is-active">
  <div class="modal-background"></div>
  <div class="modal-content" style="width: 80%;">
    <div class="box">test</div>
  </div><button class="modal-close is-large" aria-label="close" id="closeButton"></button>
</div>
`;
